io/io/inc/ROOT/RRawFile.hxx

```diff
@@ -36,7 +36,9 @@ namespace Internal {
  * opened when required (on reading, getting file size) and closed on object destruction.
  *
  * RRawFiles manage system resources and are therefore made non-copyable. They can be explicitly cloned though.
- * RRawFile are not thread-safe.
+ *
+ * RRawFile objects are not thread-safe but conditionally safe / thread-friendly: different objects can be used
+ * by different threads concurrently and the same object can be used from different threads if protected by a mutex.
```

Comment:
Because of the fFilePos this is not entirely accurate.  For example, assuming that every single statement is protected by the same mutex:
```
"on thread 1" :   rawfile->Seek(pos1);
"on thread 2":   rawfile->Seek(pos2);
"on thread 1":   rely on the seek, forexample
                          assert(rawfile->GetFilePos() == pos1);
```
I.e. Seek and any operation that relies on it **must** be within the same mutex/lock/critical-section and consequently either fFilePos's usefullness is dubvious or the rawfile object can not be shared at all by threads.

Reply:
I don't see how that contradicts "conditionally safe", as `fFilePos` is not a static?

Reply:
My example is addressing:

> the same object can be used from different threads if protected by a mutex.

where the "protected by a mutex" is not enough guidance in the sense that it does not mean "with each call protected by a mutex" but rather "with related calls protected as a whole by a mutex" and then the "related calls" need to be clarified.

Reply:
Hm, I don't quite see that this requires additional commenting.  Aren't we essentially saying: calls that change the state change the state?  The calls might come from different threads.  When different threads use the same object, it always is the case that we cannot know the ordering of the calls upfront.

Reply:
> calls that change the state change the state?

But not all 'state' are created equal :).   Many of the interface are (seemingly) "atomic" (i.e. get the work done if you lock just that function and assume other thread doesn't close/delete the file) while the Seek/SetFilePos is not (it always need to be locked with its subsequent Read/Write call).

If the RRawFile object is going to be used by several threads, I am not seeing the advantage of caching the last-seeked position.  (The only gain is the (unlikely) avoidance of seek by the pattern:  lock, check GetFilePos, "Seek if not where we want to be", do operation.

I am arguing that we either
(a) claim that the RRawFile are to be used only be a single threads/user  [conversely is there really a valid use case for sharing a RRawFile given the current interfaces]
or
(b) Remove Seek/GetFilePos and have the position always part of the call (hence, locking can always be limited to one call) [or maybe there is more odd state]
or
(c) document the parts/state/functions that can be inadvertently changed (position is one, but does 2 distinct calls to Map invalidate each other)?